twake/frontend/src/app/services/workspaces/workspaces_apps.js

```diff
@@ -292,16 +292,19 @@ class WorkspacesApps extends Observable {
   }
 
   getAppIcon(app, feather = false) {
-    if (app.simple_name.toLocaleLowerCase() == 'twake_calendar') {
-      return feather ? Calendar : 'calendar-alt';
-    }
-    if (app.simple_name.toLocaleLowerCase() == 'twake_drive') {
-      return feather ? Folder : 'folder';
-    }
-    if (app.simple_name.toLocaleLowerCase() == 'twake_tasks') {
-      return feather ? CheckSquare : 'check-square';
+    if (app) {
+      if (app.simple_name.toLocaleLowerCase() == 'twake_calendar') {
```

Comment:
Even if it can be done with a switch, if you want to keep so many if, simplify it like:

```js
const appName = app.simple_name. toLocaleLowerCase();

if (appName === 'twake_calendar') {
  return return feather ? Calendar : 'calendar-alt';
}
// etc
```

Reply:
> 1. What happens if the tab name is lonnnnnnnnngggggggg. Do we have a limit and an ellipsis?
> 2. Can we drag and drop the tabs to change order? (I guess no, but will be nice to, maybe later)

1. I am for limiting size on tab names @stephanevieira75 
2. We put an "order" field in the tabs resource model, so we will be able to do it in the future! But not now, we could have a "move left" "move right" button but I don't want to think about drag-drop so soon, may be next year :)

Reply:
Updated the getAppIcon function, added an ellipsis for the tab name and  max length for the input. :smiley: